### PR TITLE
fix missing last waypoint in CUP Task

### PR DESCRIPTION
**Pull request by AlphaLima**
-simplified CUP task load
-remove doubled waypoints
-prefer airfields over waypoint if double defined (GA)
according to this bug report:
https://www.postfrontal.com/forum/topic.asp?whichpage=1&TOPIC_ID=9115&#72126

### CODE REVIEW

File: Common/Source/SaveLoadTask/LoadCupTask.cpp [hunk 1]
```diff
@@ -257,104 +257,51 @@ class CupObsZoneUpdater {
...
+
+    for (uint i =0; i < Entries.size() ; i++)
+    {
+      mapCode2Waypoint_t::iterator It = mapWaypoint.find(Entries[Idx++].c_str());
+      if(It != mapWaypoint.end())
+      {
+        int ix = FindOrAddWaypoint(&(It->second),true);  // first try to find airfield
+        if(ix < 0)
+          ix= FindOrAddWaypoint(&(It->second),true); // not found try if waypoint exist
+
```
Comment (brunotl) [inline Common/Source/SaveLoadTask/LoadCupTask.cpp:274]: line 271 and 273 are indentical so 272 and 273 can be removed

File: Common/Source/SaveLoadTask/LoadCupTask.cpp [hunk 2]
```diff
@@ -257,104 +257,51 @@ class CupObsZoneUpdater {
...
+#ifdef DEBUG_TASK_LOAD
+#include "Globals.h"
+  int i =0;
+  while ( Task[i].Index  >=0)
+  {
+    StartupStore(_T(".......Taskpoint: %u %s\n"),(unsigned)(i), WayPointList[Task[i].Index].Name );
+    i++;
   }
-   return bLoadComplet;
+#endif
```
Comment (brunotl) [inline Common/Source/SaveLoadTask/LoadCupTask.cpp:297]: line 289 to 298 are useless, we have debugger ...
worst, including Global.h inside function body is big mistake, it's undefined behavior ...

File: Common/Source/SaveLoadTask/LoadCupTask.cpp [hunk 3]
```diff
@@ -257,104 +257,51 @@ class CupObsZoneUpdater {
...
+
+
+if(idxTP > 0)
+  return true;
+else
+  return false;
```
Comment (brunotl) [inline Common/Source/SaveLoadTask/LoadCupTask.cpp:303]: could be simply :
```cpp
return (idxTP > 0);
```

File: Common/Source/SaveLoadTask/LoadCupTask.cpp (hunk 1 above)
Comment (AlphaLima) [reply Common/Source/SaveLoadTask/LoadCupTask.cpp:274]: yes, was a mistake commig from copying already fixed in a new commit.

File: Common/Source/SaveLoadTask/LoadCupTask.cpp (hunk 2 above)
Comment (AlphaLima) [reply Common/Source/SaveLoadTask/LoadCupTask.cpp:297]: way already disabled, with the second commit

File: Common/Source/SaveLoadTask/LoadCupTask.cpp (hunk 3 above)
Comment (AlphaLima) [reply Common/Source/SaveLoadTask/LoadCupTask.cpp]: done

**Comment by AlphaLima** [comment]
squashed and pushed with another commit

**Pull request closed without merging**